src/pkg/repository/repository.go

```diff
@@ -288,3 +302,45 @@ func (r repository) DeleteBackup(ctx context.Context, id model.StableID) error {
 
 	return sw.DeleteBackup(ctx, id)
 }
+
+// ---------------------------------------------------------------------------
+// Repository ID Model
+// ---------------------------------------------------------------------------
+
+// repositoryModel identifies the current repository
+type repositoryModel struct {
+	model.BaseModel
+}
+
+const (
+	repositoryID = "repository-id"
+	repoIDTag    = "repo-id"
+)
+
+// should only be called on init.
+func newRepoModel(repoID string) repositoryModel {
+	return repositoryModel{
+		BaseModel: model.BaseModel{
+			ID: repositoryID,
+			Tags: map[string]string{
+				repoIDTag: repoID,
+			},
+		},
+	}
+}
+
+// retrieves the repository info
+func getRepoModel(ctx context.Context, ms *kopia.ModelStore) (*repositoryModel, error) {
+	rm := &repositoryModel{}
+	return rm, ms.Get(ctx, model.RepositorySchema, repositoryID, rm)
```

Comment:
I would prefer storing the unique generated ID in the ID field of the model and doing `GetIDsForType` to lookup the model that we need. As `GetIDsForType` returns `[]*BaseModel`, it's still possible to extract the ID without more lookups. This would also give us a way to ensure only a single `repositoryModel` is added to the model store

Since there should only be one `repositoryModel` in the model store, it should be clear which entry in the returned slice to use

Reply:
I dig it.